src/prompt.ts

```diff
@@ -1,170 +1,276 @@
- * have a value in `options` will cause an error to be returned. Once the values
- * are queried, the values for them are put onto the `options` object, and the
- * answers are returned.
- * @param options The options object passed through by Command.
- * @param questions `Question`s to ask the user.
- * @return The answers, keyed by the `name` of the `Question`.
- */
-export async function prompt(
-  options: Options,
-  // NB: If Observables are to be added here, the for loop below will need to
-  // be adjusted as well.
-  questions: ReadonlyArray<inquirer.DistinctQuestion>,
-): Promise<any> {
-  const prompts = [];
-  // For each of our questions, if Options already has an answer,
-  // we go ahead and _skip_ that question.
-  for (const question of questions) {
-    if (question.name && options[question.name] === undefined) {
-      prompts.push(question);
+/**
+ * Options for the Input function.
+ *
+ * Exported because Inqurier does not export its own input configs anymore. Some
+ * unused options are missing, such as theme.
+ */
+export type InputConfig = BasicOptions<string> & {
+  transformer?: (
+    value: string,
+    {
+      isFinal,
+    }: {
+      isFinal: boolean;
+    },
+  ) => string;
+  validate?: (value: string) => boolean | string | Promise<string | boolean>;
+};
+
+/**
+ * Prompt for a string input.
+ */
+export async function input(message: string): Promise<string>;
+/**
+ * Prompt for a string input.
+ */
+export async function input(opts: InputConfig): Promise<string>;
+
+export async function input(opts: InputConfig | string): Promise<string> {
+  if (typeof opts === "string") {
+    opts = { message: opts };
+  } else {
+    const { shouldReturn, value } = guard(opts);
+    if (shouldReturn) {
+      return value;
     }
   }
+  return inquirer.input(opts);
+}
 
-  if (prompts.length && options.nonInteractive) {
-    const missingOptions = Array.from(new Set(prompts.map((p) => p.name))).join(", ");
-    throw new FirebaseError(
-      `Missing required options (${missingOptions}) while running in non-interactive mode`,
-      {
-        children: prompts,
-      },
-    );
+/**
+ * Options for the confirm function.
+ *
+ * Epxorted because Inquirer does not export its own input configs anymore. Some unused
+ * options are missing, such as theme.
+ */
+export type ConfirmConfig = BasicOptions<boolean> & {
+  transformer?: (value: boolean) => string;
+};
+
+/**
+ * Prompt a user to confirm a selection
+ */
+export async function confirm(message: string): Promise<boolean>;
+/**
+ * Prompt a user to confirm a selection
+ * Will abort if nonInteractive and not force
+ */
+export async function confirm(opts: ConfirmConfig): Promise<boolean>;
+
+export async function confirm(opts: string | ConfirmConfig) {
+  if (typeof opts === "string") {
+    opts = { message: opts };
+  } else {
+    if (opts.force) {
+      // TODO: Should we print what we've forced?
```

Comment:
Good question - I like the idea (especially since it gives a 'paper trail' when running noninteractive commands). We probably need to audit the places we use this tho to make sure it doesn't lead to too many redundant logs